.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,26 @@
+# Run tests when merged to main branch
+
+name: Run all the tests
+on:
+  push:
+    branches:
+      - main
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref_name }}
```

Comment:
We don't need to checkout the whole branch, this option can be omitted. Without this option, GitHub will automatically use the latest commit